examples/linear_model/plot_nnls.py

```diff
@@ -0,0 +1,68 @@
+"""
+==========================
+Non-negative least squares
+==========================
+
+In this example, we fit a linear model with positive constraints on the
+regression coefficients and compare the estimated coefficients to a classic
+linear regression.
+"""
+print(__doc__)
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.metrics import r2_score
+
+# %%
+# Generate some random data
+np.random.seed(42)
+
+n_samples, n_features = 200, 50
+X = np.random.randn(n_samples, n_features)
+true_coef = 3 * np.random.randn(n_features)
+# Threshold coefficients to render them non-negative
+true_coef[true_coef < 0] = 0
+y = np.dot(X, true_coef)
+
+# Add some noise
+y += 5 * np.random.normal(size=(n_samples, ))
+
+# %%
+# Split the data in train set and test set
+from sklearn.model_selection import train_test_split
```

Comment:
We like to have self contained blocks for the feature highlights examples, but for regular examples I think we still import everything at the beginning?

It's the recommended practice in general. Also the import below ` from sklearn.linear_model import LinearRegression ` is on a block of its own and it's a little bit strange.

Reply:
> We like to have self contained blocks for the feature highlights examples, but for regular examples I think we still import everything at the beginning?

It wasn't my understanding for my previous contributions to examples. I was told to start to write blocks to emulate jupyter-notebook structure (as the notebook version is downloadable), maybe I've misunderstood... @GaelVaroquaux , @glemaitre?

> Also the import below `from sklearn.linear_model import LinearRegression` is on a block of its own and it's a little bit strange.

Agree, I have moved the import in the next 'cell'.

Reply:
I'd say that notebook style doesn't necessarily mean "only import when you need it" but I can see the value of doing so. The `r2_score` import at the top is a bit of an outlier here ;)

It's not super important anyway so I'll merge (I pushed an empty commit to trigger the CI, which previously didn't finish)

Thanks for the work @cmarmo!

Reply:
Indeed, the r2_score import isn't at the right place... but you already merged it... ;)
Thanks for your understanding!